packages/firebase_data_connect/firebase_data_connect/lib/src/optional.dart

```diff
@@ -65,11 +65,16 @@ class Optional<T> {
   }
 }
 
-String nativeToJson<T>(T type) {
-  if (type is bool || type is int || type is double || type is num) {
-    return type.toString();
-  } else if (type is String) {
+dynamic nativeToJson<T>(T type) {
+  if (type is bool ||
+      type is int ||
+      type is double ||
+      type is num ||
+      type is String) {
     return type;
```

Comment:
Are you sure this is correct? If we're making a type a string, wouldn't we need to keep:

```dart
if (type is bool || type is int || type is double || type is num) {
    return type.toString();
  }
```

Reply:
I modified the serializers to accept dynamic return types because of the `AnyScalar`, which allows you to pass in any json-serializable type. If we allow serialization to Strings in this function, over the wire, we'll be sending `"true"` instead of `true`